### ISSUE TITLE
Make filename for single st2client file / release reusable

### ISSUE THREAD

**Issue by winem** [pypa/warehouse#10392]
**What's the problem this feature will solve?**
This will unblock us, the StackStorm maintainers (https://github.com/StackStorm/st2) in the release process. One of our scripts failed while releasing our latest release 3.6.0 of the st2client (https://pypi.org/manage/project/st2client/) and we had to trigger it again. Unfortunately it's not yet idempotent. So we fixed the issue and removed the release 3.6.0 from PyPI to ensure that our script will work without deviations on the 2nd run. Now we can't upload the 3.6.0 artifacts again.

**Describe the solution you'd like**
While we understand that there are good reasons have non-reusable filenames, it would be great and highly appreciated if there is any chance to make an exception. st2client 3.6.0 was available for less than 3 hours on saturday evening. 

![Selection_314](https://user-images.githubusercontent.com/1457899/142775832-715f97f4-ad11-418b-9cb9-c5a1a7d3bb07.png)

**Additional context**
This would be really helpful and allow us to complete the release process without having to increment the versions of all our components and restart the whole process. st2client is "just" the tool to manage and orchestrate StackStorm. The files we'll upload will be exactly the same as those that were uploaded on Saturday. The issue we fixed was a configuration issue related to the vagrant/OVA setup and occurred at a latter step in the same workflow as we use to publish the packages to PyPI. So there is no change to the code of the st2client package itself. 

It would be great if this can be done!

**Comment by ewjoachim** [comment]
If I'm not mistaken, if you do a `3.6.0.post0`, pip will recognize it as a 3.6.0 while still having a different version number. https://www.python.org/dev/peps/pep-0440/#post-releases

Would that solve the issue for you ?

**Comment by winem** [comment]
That's actually something I'm currently evaluating, too. We do know that this requires some changes on our release scripts and the CI/CD but it's a bit tricky to identify all the places where we have to take care of it in advance. 

Our biggest pain point with the current situation is that some of our components are already released and available (i.e. on dockerhub) but others like the st2client aren't. With your support we'll be able to publish our components to all platforms before EOB today, without it will probably take a few more days. 

We already created tasks to discuss .point releases in the future and make our CD scripts idempotent.

So it would still be super helpful if you can support us here.

**Comment by di** [comment]
Hi @winem, thanks for creating the issue. In the short term, we're unable to offer an exception for `st2client` project, as it's both against our general policy to make exceptions like this, and also not something we have the technical ability to easily do without making changes to PyPI and to assumptions that downstream mirrors make about consuming artifacts from PyPI.

Like @ewjoachim mentioned, post-releases should be used for exactly this kind of situation, and should work exactly the same with installers like `pip`. I'd recommend this as your best path forward here.

Long-term, resolving #4440 would satisfy this use case, so I'm going to close this issue as a duplicate of that issue.

**Comment by winem** [comment]
Hi @di, 

thanks for the feedback. I totally understand that. 

I'll see if I can contribute something to #4440.
